commit fb4cadc9cb9813a986650a758657034a10fc4cc1 (refs/changes/69/3269/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-31T16:23:01+01:00 (1 year, 8 months ago)
    
    TOOLS-1725 sdcadm up should not bypass Triton stack
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/procedures/shared.js b/lib/procedures/shared.js
index 000b9a6..82f6f5e 100644
--- a/lib/procedures/shared.js
+++ b/lib/procedures/shared.js
@@ -345,8 +345,13 @@ function waitForInstToBeUp(arg, cb) {
                         next();
                         return;
                     }
+                    // Given `sdcadm up` calls `vmadm reprovision` directly,
+                    // we need to make sure that VMAPI gets updated with the
+                    // most recent known state for the VM being reprovisioned.
+                    // Therefore the "sync=true" flag usage here:
                     sdcadm.vmapi.getVm({
-                        uuid: uuid
+                        uuid: uuid,
+                        sync: true
                     }, function (err, vm) {
                         if (err) {
                             arg.opts.log.debug({
